### DISCUSSION TITLE
`then` in try\catch

### DISCUSSION THREAD

**Discussion by JBBianchi** [serverlessworkflow/specification#914] [Ideas] [answered]
While talking on Slack with [Modood](https://cloud-native.slack.com/archives/CMBAZQSEP/p1720693190446679), a problematic appeared I don't think we have a solution for right now:
When [trying](https://github.com/serverlessworkflow/specification/blob/main/dsl-reference.md#try) a (set of)task(s), we can `catch` some errors and either retry using a [retryPolicy ](https://github.com/serverlessworkflow/specification/blob/main/dsl-reference.md#retry) or execute some other tasks with `do`. But unless I missed something, we cannot gracefully transition using `then`. 

Shouldn't we introduce a flow directive in the catch block ?

@cdavernas @fjtirado @matthias-pichler-warrify @ricardozanini wdyt ?

**Comment by cdavernas** [comment, marked as answer]
IMHO it's a good bad idea: `try/catch` is not a `switch` (aka flow control).
To do what you propose, you can do something in the `do`, then `switch`. Having a `then` at top level plus one in the catch clause would be IMO confusing, in addition of adding a new flow control that WILL be misused (ex: throwing an error to explicitly flow using the catch clause).
The `try` clause is not intended, in any language I know of, to be a flow directive. It's a construct to handle an error then continue/resume the normal flow.

**Reply by JBBianchi** [reply]
My bad, I skipped the top-level `then`. Nevermind
